power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -169,8 +169,12 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
     using ComponentFlags = std::array<bool, n_types>;
     using ComponentCountInBase = std::pair<std::string, Idx>;
 
+    // TODO: (figueroa1395) probably move to common.hpp or somewhere else
```

Comment:
`invalid_index` will probably be moved to `.../main_core/update.hpp` since it is only going to be used there after the last step of refactoring. The rest are only used in `main_model_impl.hpp` (at least for now), so I will leave them be until the bigger refactoring happens, as they may be moved around or removed alltogether later.